Libraries/Components/WebView/WebView.ios.js

```diff
@@ -462,6 +462,38 @@ var WebView = React.createClass({
   },
 
   /**
+   * Evaluate JavaScript on the current page.
+   */
+  evaluateJavaScript: function(
+    script: string,
+    callback?: ?(error: ?Error, result: ?string) => void,
+  ): Promise {
+    return new Promise((resolve, reject) => {
+      var escaped = JSON.stringify(script).replace(/\u2028/g, '\\u2028').replace(/\u2029/g, '\\u2029');
+      var wrapped = 'JSON.stringify(eval(' + escaped + '))';
+
+      RCTWebViewManager.evaluateJavaScript(this.getWebViewHandle(), wrapped, function(error, result) {
+        if (!error) {
+          if (result === '') {
+            error = new Error("Error evaluating script.")
```

Comment:
Unfortunately, `stringByEvaluatingJavaScriptFromString` simply returns `nil` if an error occurs. You can still see the complete error on the page using the remote debugger (as you would if you were using this API natively).

One idea I did try initially was wrapping the script in a `try`/`catch` block, returning a custom error object if an exception is thrown. But that then conceals the error from the page, making it difficult to break/inspect the error using the remote debugger. I think it makes sense to keep the error page-side since that's the context the script is being evaluated in.

Note that anyone using this API can always add the `try`/`catch` wrapper themselves to the script if they'd prefer to catch the error in React instead.